Expensify/App
# Return promise in payIOUReport / Fix AdditionalDetailsStep

cc @nickmurray47 @aldo-expensify 

### Details

- Return promise in payIOUReport and remove `shouldRedirectToChatReport` param
- Also fixes the `AdditionalDetailsStep` which I noticed was broken by [this PR](https://github.com/Expensify/App/pull/6569) when testing 

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/190539

### Tests
Switch to https://github.com/Expensify/Web-Expensify/pull/32181

1. Add personal bank account and pass KYC checks by following -> https://stackoverflow.com/c/expensify/questions/10607
2. Attempt to send money to someone else
3. Verify you are navigated back to the current report on success

### QA Steps

None yet as these flows are under beta

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
